Simplenote/NoticeController.swift

```diff
@@ -0,0 +1,114 @@
+import UIKit
+
+class NoticeController {
+    // MARK: Proerties
+    //
+    static let shared = NoticeController()
+
+    private var notices: [Notice] = []
+    private var current: Notice?
+    private let noticePresenter = NoticePresenter()
+    private let timerFactory: TimerFactory
+
+    private var timer: Timer? {
+        didSet {
+            oldValue?.invalidate()
+        }
+    }
+
+    private var isPresenting: Bool {
+        current != nil
+    }
+
+    // MARK: Life Cycle
+    //
+    private init() {
+        self.timerFactory = TimerFactory()
+    }
+
+    func setupNoticeController() {
+        noticePresenter.startListeningToKeyboardNotifications()
+    }
+
+    // MARK: Presenting
+    //
+    func present(_ notice: Notice) {
+        if isPresenting {
+            appendToQueueIfNew(notice)
+            return
+        }
+
+        current = notice
+        let noticeView = makeNoticeView(from: notice)
+
+        noticePresenter.presentNoticeView(noticeView) { () in
+            let delay = self.current?.action == nil ? Times.shortDelay : Times.longDelay
+            self.timer = self.timerFactory.scheduledTimer(with: delay, completion: {
+                self.dismiss()
+            })
+        }
+    }
+
+    /// Confirms if a notice is already contained in notices queue. Appends to queue if new
+    ///
+    private func appendToQueueIfNew(_ notice: Notice) {
+        if notices.contains(notice) {
+            return
+        }
+
+        if notice == current {
+            return
+        }
+
+        notices.append(notice)
+    }
+
+    private func makeNoticeView(from notice: Notice) -> NoticeView {
+        let noticeView: NoticeView = NoticeView.instantiateFromNib()
+        noticeView.message = notice.message
+        noticeView.actionTitle = notice.action?.title
+        noticeView.handler = notice.action?.handler
+        noticeView.delegate = self
+
+        return noticeView
+    }
+
+    // MARK: Dismissing
+    //
+    @objc
+    func dismiss() {
```

Comment:
I had left this as internal so that it could be tested. I noticed in your comments on the testing PR, there were some suggestions on how we can test this and leave it private.  I will handle that there